hsphfpd.pl

```diff
@@ -485,7 +485,7 @@ sub hsphfpd_our_battery_level {
 				hsphfpd_csr_send_battery_level($endpoint);
 			}
 			if (exists $endpoints{$endpoint}->{apple_features}->{'apple-battery-level'} and $our_battery_level != -1) {
-				hsphfpd_socket_write($endpoint, "AT+IPHONEACCEV=1,1," . int(9 * $our_battery_level / 100 + 0.5));
+				hsphfpd_socket_write($endpoint, "AT+IPHONEACCEV=1,1," . int($our_battery_level / 10 - 1));
```

Comment:
This formula is suspicious...
```
$ perl -E 'say "pecent=$_" . " level=" . int($_/10-1) foreach 0..100'
pecent=0 level=-1
pecent=1 level=0
pecent=2 level=0
pecent=3 level=0
pecent=4 level=0
pecent=5 level=0
pecent=6 level=0
pecent=7 level=0
pecent=8 level=0
pecent=9 level=0
pecent=10 level=0
pecent=11 level=0
pecent=12 level=0
pecent=13 level=0
pecent=14 level=0
pecent=15 level=0
pecent=16 level=0
pecent=17 level=0
pecent=18 level=0
pecent=19 level=0
pecent=20 level=1
pecent=21 level=1
pecent=22 level=1
pecent=23 level=1
pecent=24 level=1
pecent=25 level=1
pecent=26 level=1
pecent=27 level=1
pecent=28 level=1
pecent=29 level=1
pecent=30 level=2
pecent=31 level=2
pecent=32 level=2
pecent=33 level=2
pecent=34 level=2
pecent=35 level=2
pecent=36 level=2
pecent=37 level=2
pecent=38 level=2
pecent=39 level=2
pecent=40 level=3
pecent=41 level=3
pecent=42 level=3
pecent=43 level=3
pecent=44 level=3
pecent=45 level=3
pecent=46 level=3
pecent=47 level=3
pecent=48 level=3
pecent=49 level=3
pecent=50 level=4
pecent=51 level=4
pecent=52 level=4
pecent=53 level=4
pecent=54 level=4
pecent=55 level=4
pecent=56 level=4
pecent=57 level=4
pecent=58 level=4
pecent=59 level=4
pecent=60 level=5
pecent=61 level=5
pecent=62 level=5
pecent=63 level=5
pecent=64 level=5
pecent=65 level=5
pecent=66 level=5
pecent=67 level=5
pecent=68 level=5
pecent=69 level=5
pecent=70 level=6
pecent=71 level=6
pecent=72 level=6
pecent=73 level=6
pecent=74 level=6
pecent=75 level=6
pecent=76 level=6
pecent=77 level=6
pecent=78 level=6
pecent=79 level=6
pecent=80 level=7
pecent=81 level=7
pecent=82 level=7
pecent=83 level=7
pecent=84 level=7
pecent=85 level=7
pecent=86 level=7
pecent=87 level=7
pecent=88 level=7
pecent=89 level=7
pecent=90 level=8
pecent=91 level=8
pecent=92 level=8
pecent=93 level=8
pecent=94 level=8
pecent=95 level=8
pecent=96 level=8
pecent=97 level=8
pecent=98 level=8
pecent=99 level=8
pecent=100 level=9
```

Reply:
Definitely. Well, I just woke up.

What about this: `int($_ >= 5 ? $_/10-0.5 : 0)` ? It rounds to the nearest decade, except that the [0-5) range would be the same as [5-15), but at least that's consistent with the rest of the PR.

Reply:
```perl -E 'say "pecent=$_" . " level=" . int($_ >= 5 ? $_/10-0.5 : 0) foreach 0..100'
pecent=0 level=0
pecent=1 level=0
pecent=2 level=0
pecent=3 level=0
pecent=4 level=0
pecent=5 level=0
pecent=6 level=0
pecent=7 level=0
pecent=8 level=0
pecent=9 level=0
pecent=10 level=0
pecent=11 level=0
pecent=12 level=0
pecent=13 level=0
pecent=14 level=0
pecent=15 level=1
pecent=16 level=1
pecent=17 level=1
pecent=18 level=1
pecent=19 level=1
pecent=20 level=1
pecent=21 level=1
pecent=22 level=1
pecent=23 level=1
pecent=24 level=1
pecent=25 level=2
pecent=26 level=2
pecent=27 level=2
pecent=28 level=2
pecent=29 level=2
pecent=30 level=2
pecent=31 level=2
pecent=32 level=2
pecent=33 level=2
pecent=34 level=2
pecent=35 level=3
pecent=36 level=3
pecent=37 level=3
pecent=38 level=3
pecent=39 level=3
pecent=40 level=3
pecent=41 level=3
pecent=42 level=3
pecent=43 level=3
pecent=44 level=3
pecent=45 level=4
pecent=46 level=4
pecent=47 level=4
pecent=48 level=4
pecent=49 level=4
pecent=50 level=4
pecent=51 level=4
pecent=52 level=4
pecent=53 level=4
pecent=54 level=4
pecent=55 level=5
pecent=56 level=5
pecent=57 level=5
pecent=58 level=5
pecent=59 level=5
pecent=60 level=5
pecent=61 level=5
pecent=62 level=5
pecent=63 level=5
pecent=64 level=5
pecent=65 level=6
pecent=66 level=6
pecent=67 level=6
pecent=68 level=6
pecent=69 level=6
pecent=70 level=6
pecent=71 level=6
pecent=72 level=6
pecent=73 level=6
pecent=74 level=6
pecent=75 level=7
pecent=76 level=7
pecent=77 level=7
pecent=78 level=7
pecent=79 level=7
pecent=80 level=7
pecent=81 level=7
pecent=82 level=7
pecent=83 level=7
pecent=84 level=7
pecent=85 level=8
pecent=86 level=8
pecent=87 level=8
pecent=88 level=8
pecent=89 level=8
pecent=90 level=8
pecent=91 level=8
pecent=92 level=8
pecent=93 level=8
pecent=94 level=8
pecent=95 level=9
pecent=96 level=9
pecent=97 level=9
pecent=98 level=9
pecent=99 level=9
pecent=100 level=9```

Reply:
Would not it be better to have following calculation?
0%..10% -- level 0
11%..20% -- level 1
...
91%..100% -- level 9

There would be less percent points with zero level value.

Reply:
Honestly, that's a matter of personal taste. Like this, then: `int($_ > 0 ? ($_-1)/10 : 0)` ? I'll make the change.

Reply:
No need for special check for zero. `int(-1/10)` is also zero.

Reply:
Oh, right.